sys/key_rev_archival_value.js

```diff
@@ -0,0 +1,149 @@
+"use strict";
+
+var P = require('bluebird');
+var uuid = require('cassandra-uuid').TimeUuid;
+var preq = require('preq');
+
+var HyperSwitch = require('hyperswitch');
+var HTTPError = HyperSwitch.HTTPError;
+var URI = HyperSwitch.URI;
+
+var mwUtil = require('../lib/mwUtil');
+
+var spec = HyperSwitch.utils.loadSpec(__dirname + '/key_rev_value.yaml');
+
+function ArchivalBucket(options) {
+}
+
+ArchivalBucket.prototype._latestName = function(bucket) {
+    return bucket + '.latest';
+};
+
+ArchivalBucket.prototype._archiveName = function(bucket) {
+    return bucket + '.archive';
+};
+
+ArchivalBucket.prototype.createBucket = function(hyper, req) {
+    var self = this;
+    var rp = req.params;
+    return P.join(
+        hyper.put({
+            uri: new URI([rp.domain, 'sys', 'key_value', self._latestName(rp.bucket)]),
+            headers: req.headers,
+            body: req.body
+        }),
+        hyper.put({
+            uri: new URI([rp.domain, 'sys', 'key_rev_value', self._archiveName(rp.bucket)]),
+            headers: req.headers,
+            body: req.body
+        })
+    )
+    .then(function() {
+        return { status: 201 }; });
+};
+
+ArchivalBucket.prototype.getRevision = function(hyper, req) {
+    var self = this;
+    var rp = req.params;
+    var latestRequestPath = [rp.domain, 'sys', 'key_value', self._latestName(rp.bucket), rp.key];
+    if (rp.tid) {
+        latestRequestPath.push(rp.tid);
+    }
+    return hyper.get({
+        uri: new URI(latestRequestPath),
+        headers: req.headers
+    })
+    .then(function(res) {
+        if (rp.revision) {
+            var etagInfo = mwUtil.parseETag(res.headers.etag);
+            if (Number.parseInt(etagInfo.rev) !== Number.parseInt(rp.revision)
```

Comment:
Very minor: `parseInt` is global, so the `Number` prefix shouldn't be needed here.